packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -271,8 +276,17 @@ function WorklistDataGrid(props) {
     if (currentSelectedCell != null) {
       setCurrentSelectedCell(null);
     }
+
+    if (!hasSelectableRows) {
+      rowSelection.current.focussedRow = null;
+      rowSelection.current.lastIndividuallySelectedRow = null;
+    }
+
     // Since the row selection mode has changed, the row selection mode needs to be updated.
-    setAriaLiveMessage(intl.formatMessage({ id: hasSelectableRows ? 'Terra.worklist-data-grid.row-selection-mode-enabled' : 'Terra.worklist-data-grid.row-selection-mode-disabled' }));
+    if (!rowSelection.current.isRowSelectionEnabledByGrid) {
+      setAriaLiveMessage(intl.formatMessage({ id: hasSelectableRows ? 'Terra.worklist-data-grid.row-selection-mode-enabled' : 'Terra.worklist-data-grid.row-selection-mode-disabled' }));
```

Comment:
why do we care if a row selection was toggled by the grid?  It really isn't clear what the purpose of this variable is by looking at the name.

Reply:
In the current implementation, when the user turns on row selection mode using the grid's keyboard shortcuts, the announcement is being made before the consumer is notified. This check prevents a repetition of the same announcement after the state change by the consumer. The UXPLATFORM-9507 story will change all announcements to come after the consumer changes the state. It will make this variable & check no longer necessary but such change should be part of that story.

Reply:
So should we just remove those messages for now or in the next JIRA to reduce the complexity of what we are reviewing and have to update in the subsequent JIRA?  I question if it is worth the extra effort to manage messages that we know are going away.

Reply:
The code matches the current requirement. All changes should be addressed holistically by the story that has the new requirements.